spec/text-buffer-spec.coffee

```diff
@@ -990,14 +991,16 @@ describe "TextBuffer", ->
 
       it "notifies ::onDidConflict observers", ->
         buffer.setText("a change")
-        buffer.save()
-        buffer.insert([0, 0], "a second change")
-
         handler = jasmine.createSpy('fileChange')
-        writeFileSync(filePath, "a disk change")
-        buffer.onDidConflict handler
+        buffer.save().then ->
+          buffer.insert([0, 0], "a second change")
+          buffer.onDidConflict handler
+          expect(handler.callCount).toBe 0
+          # wait a little bit to not confuse the save change event, with the write file change event
+          setTimeout ( ->
+            writeFileSync(filePath, "a disk change")
+          ), 500
```

Comment:
every save emits a watcher change and if two saves happens really fast:
* The first is a good normal save, triggered by the IDE
* The second is conflicting non-editor-related conflict that happened almost the same time the edited the text buffer, then it's a conflict - but the change watcher was emitted twice almost the same time and checked and emitted two conflict events for one real conflict.

I was actually waiting on your help here to get this right, because this didn't happen when the save was synchronous.